core/src/main/java/com/datastax/oss/driver/api/core/data/ByteUtils.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.api.core.data;
+
+import com.datastax.oss.protocol.internal.util.Bytes;
+import java.nio.ByteBuffer;
+
+/**
+ * A set of static utility methods to work with byte buffers (associated with CQL type {@code
+ * blob}).
+ */
+public class ByteUtils {
```

Comment:
I'm still seeing `Bytes` being used in a lot of places in the java driver:

```
        java-driver-core  (24 usages found)
            com.datastax.oss.driver.internal.core.cql  (5 usages found)
                Conversions  (3 usages found)
                    toMessage(Statement<?>, DriverExecutionProfile, InternalDriverContext)  (3 usages found)
                        191 Bytes.getArray(id),
                        192 (resultMetadataId == null) ? null : Bytes.getArray(resultMetadataId),
                        222 queriesOrIds.add(Bytes.getArray(boundStatement.getPreparedStatement().getId()));
                CqlRequestHandler.NodeResponseCallback  (1 usage found)
                    processErrorResponse(Error)  (1 usage found)
                        655 Bytes.toHexString(id)));
                DefaultRow.SerializationProxy  (1 usage found)
                    SerializationProxy(DefaultRow)  (1 usage found)
                        154 this.values[i] = (buffer == null) ? null : Bytes.getArray(buffer);
            com.datastax.oss.driver.internal.core.data  (2 usages found)
                DefaultTupleValue.SerializationProxy  (1 usage found)
                    SerializationProxy(DefaultTupleValue)  (1 usage found)
                        187 this.values[i] = (buffer == null) ? null : Bytes.getArray(buffer);
                DefaultUdtValue.SerializationProxy  (1 usage found)
                    SerializationProxy(DefaultUdtValue)  (1 usage found)
                        204 this.values[i] = (buffer == null) ? null : Bytes.getArray(buffer);
            com.datastax.oss.driver.internal.core.metadata.schema.parsing  (2 usages found)
                DataTypeClassNameParser  (1 usage found)
                    parse(CqlIdentifier, String, Map<CqlIdentifier, UserDefinedType>, InternalDriverContext)  (1 usage found)
                        107 Bytes.fromHexString("0x" + parser.readOne()), context.getProtocolVersion());
                DataTypeClassNameParser.Parser  (1 usage found)
                    getNameAndTypeParameters()  (1 usage found)
                        269 Bytes.fromHexString("0x" + bbHex), DefaultProtocolVersion.DEFAULT);
            com.datastax.oss.driver.internal.core.metadata.token  (6 usages found)
                ByteOrderedToken  (3 usages found)
                    ByteOrderedToken(ByteBuffer)  (1 usage found)
                        33 this.value = ByteBuffer.wrap(Bytes.getArray(value)).asReadOnlyBuffer();
                    compareTo(Token)  (1 usage found)
                        63 .compare(Bytes.getArray(value), Bytes.getArray(((ByteOrderedToken) other).value));
                    toString()  (1 usage found)
                        68 return "ByteOrderedToken(" + Bytes.toHexString(value) + ")";
                ByteOrderedTokenFactory  (2 usages found)
                    parse(String)  (1 usage found)
                        51 ByteBuffer value = Bytes.fromHexString(tokenString);
                    format(Token)  (1 usage found)
                        59 return Bytes.toHexString(((ByteOrderedToken) token).getValue());
                ByteOrderedTokenRange  (1 usage found)
                    toBigInteger(ByteBuffer, int)  (1 usage found)
                        108 byte[] bytes = Bytes.getArray(bb);
            com.datastax.oss.driver.internal.core.session  (1 usage found)
                ReprepareOnUp  (1 usage found)
                    gatherPayloadsToReprepare()  (1 usage found)
                        176 Bytes.toHexString(payload.id));
            com.datastax.oss.driver.internal.core.type.codec  (8 usages found)
                BlobCodec  (2 usages found)
                    format(ByteBuffer)  (1 usage found)
                        68 return (value == null) ? "NULL" : Bytes.toHexString(value);
                    parse(String)  (1 usage found)
                        76 : Bytes.fromHexString(value);
                CqlDurationCodec  (1 usage found)
                    decode(ByteBuffer, ProtocolVersion)  (1 usage found)
                        90 DataInput in = ByteStreams.newDataInput(Bytes.getArray(bytes));
                CustomCodec  (2 usages found)
                    format(ByteBuffer)  (1 usage found)
                        75 return (value == null) ? "NULL" : Bytes.toHexString(value);
                    parse(String)  (1 usage found)
                        83 : Bytes.fromHexString(value);
                InetCodec  (1 usage found)
                    decode(ByteBuffer, ProtocolVersion)  (1 usage found)
                        69 return InetAddress.getByAddress(Bytes.getArray(bytes));
                StringCodec  (1 usage found)
                    decode(ByteBuffer, ProtocolVersion)  (1 usage found)
                        77 return new String(Bytes.getArray(bytes), charset);
                VarIntCodec  (1 usage found)
                    decode(ByteBuffer, ProtocolVersion)  (1 usage found)
                        63 return (bytes == null) || bytes.remaining() == 0 ? null : new BigInteger(Bytes.getArray(bytes));
        java-driver-examples  (4 usages found)
            com.datastax.oss.driver.examples.json.codecs  (2 usages found)
                JacksonJsonCodec  (1 usage found)
                    decode(ByteBuffer, ProtocolVersion)  (1 usage found)
                        111 return objectMapper.readValue(Bytes.getArray(bytes), toJacksonJavaType());
                Jsr353JsonCodec  (1 usage found)
                    decode(ByteBuffer, ProtocolVersion)  (1 usage found)
                        152 try (ByteArrayInputStream bais = new ByteArrayInputStream(Bytes.getArray(bytes))) {
            com.datastax.oss.driver.examples.paging  (2 usages found)
                ForwardPagingRestUi.UserService  (2 usages found)
                    getUserVideos(int, String)  (2 usages found)
                        221 statementBuilder.setPagingState(Bytes.fromHexString(page));
                        225 String nextPage = Bytes.toHexString(rs.getExecutionInfo().getPagingState());
```
There are also usages in test classes.
There are also usages in the DSE driver, we will have to keep in mind to change that as well.

Reply:
I don't think we need to change those. The goal is to give 3rd-party users something they can rely on without importing from an `internal` package. But from our own code, using the internal class is fine.